### ISSUE TITLE
Error during compilation - 'stdlib.h' file not found

### ISSUE THREAD

**Issue by tjex** [nvim-treesitter/nvim-treesitter#3782]
### Describe the bug

Installing parsers via `:TSInstall <lang>` or when automatically run after opening file results in the below error output:
```
nvim-treesitter[lua]: Error during compilation                                                                                           
In file included from src/parser.c:1:
./src/tree_sitter/parser.h:10:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
In file included from src/scanner.c:1:
./src/tree_sitter/parser.h:10:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
```

I have been running treesitter for two months without issue and today I ran `PackerSync`, after which the errors started occurring. 
I have tested on neovim 0.8 and 0.9. 

### To Reproduce

1. uninstall all parsers from treesitter (`TSUninstall <lang>`)
> When I ran this, I got a notification that there were multiple parsers installed, and not all by treesitter and was advised to remove them, which I did. 
2. uninstall treesitter via packer (delete `use` call for treesitter and run `PackerSync`)
3. install treesitter again via packer (using the code from readme for first time install - installs without errors)
> ```
>   use {
>       'nvim-treesitter/nvim-treesitter',
>        run = function()
>            local ts_update = require('nvim-treesitter.install').update({ with_sync = true })
>            ts_update()
>        end,
>    }
>
4. quit neovim and reopen
5. navigate to a .lua file / run `TSInstall lua`
6. The above error is presented
7. comment out all other plugin loading from init.lua except treesitter.
8. repeat steps 4 - 5. No change, error persists. 

### Expected behavior

When opening files with extensions or running `TSInstall <lang>`, the parser should install without errors like it did before running `PackerSync`. 

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter[lua]: Error during compilation                                                                                           
In file included from src/parser.c:1:
./src/tree_sitter/parser.h:10:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
In file included from src/scanner.c:1:
./src/tree_sitter/parser.h:10:10: fatal error: 'stdlib.h' file not found
#include <stdlib.h>
         ^~~~~~~~~~
1 error generated.
```

### Output of `nvim --version`

```text
NVIM v0.9.0-dev-274+g894c59ec1-dirty
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by tillman@MacBook-Pro.local

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/opt/homebrew/Cellar/neovim/HEAD-894c59e/share/nvim"

Run :checkhealth for more info
```

**Comment by clason** [comment]
Duplicate of #3760

**Comment by MrOxMasTer** [comment]
So. Reinstalled windows, thought that might be the problem, but no. As I expected. I wanted to write under a different answer, but the author closed it.

I do not know why the author of this library decided that it would work fine with clang, and unfortunately I wrote about it somewhere, but I can't find it, maybe my answer will help someone

The fact is that with `clang` and `gcc` there are different standard std libraries. In gcc, it is `stdlib.h`, and in clang something like `stdlib.h`. 

Simply put, install gcc and everything will work

**Comment by mehdiAI** [comment]
When i install treesitter in nvim and set treesitter.lua file to install some languages in  ensure_installed variable,
but i got c++ compiler and stdlib.h not found errors, then i resolved it by below approach:
- Install clang for c++ compiler and after that need to visual cpp build tool.
Links:
- https://github.com/llvm/llvm-project/releases/tag/llvmorg-18.1.8
- https://visualstudio.microsoft.com/visual-cpp-build-tools/
